woocommerce/wc-admin
# Record refunds separate from order entries

Fixes #2001 

Records refunds in their own rows so that stats are reflected on the date the refund is given and not on the original order date.

* Reflects refunded amounts in orders, products/categories, taxes, and revenue.
* Removes unused refund fields no longer needed since refunds are now recorded.
* Updates tests to include refunded amounts during a period.
* Removes refunded orders from excluded statuses since they are needed to balance out the actual refund amounts.

### Thoughts

1. There’s some code duplication between the `WC_Admin_Order` and `WC_Admin_Order_Refund` classes.  We can’t use traits since we need to support PHP 5.2.  Any other thoughts on how to clean this up?
2. Should the refund count go towards the `avg_order_value`?  `( net_total / num_orders )` - Should refunds be included in the number of orders here?  This PR assumes they go towards the net total, but don't influence the order count `( net total of all orders and refunds ) / ( number of actual orders excluding refunds )`.

### Screenshots
<img width="884" alt="Screen Shot 2019-04-22 at 4 29 13 PM" src="https://user-images.githubusercontent.com/10561050/56492986-1cea8080-6520-11e9-9583-7961666495be.png">
<img width="836" alt="Screen Shot 2019-04-22 at 5 03 31 PM" src="https://user-images.githubusercontent.com/10561050/56493107-94201480-6520-11e9-94fc-41e4c0f91648.png">


### Detailed test instructions:

1.  Create an order with multiple products, taxes, and at least one coupon.
2.  Make a couple partial refunds to that order.
3.  Make sure the refunds show up correctly in each report with negative values when the refund is added.
4.  Make sure all tests pass.
